### PR TITLE
perf(orjson): Create several options to enable orjson experimentally

### PR DESCRIPTION
**Pull request by loewenheim**
This creates the following rate-based options for enabling `orjson` in various modules:

* ~`apidocs.enable-orjson`~
* `auth.enable-orjson`
* `backup.enable-orjson`
* `event-manager.enable-orjson`
* `eventstore.enable-orjson`
* `flagpole.enable-orjson`

ref: https://github.com/getsentry/sentry/issues/68903

Edit: removed apidocs because there are considerations about formatting.

### CODE REVIEW

File: src/sentry/apidocs/hooks.py [hunk 1]
```diff
@@ -87,7 +87,7 @@ def __write_ownership_data(ownership_data: dict[ApiOwner, dict]):
         index += __get_line_count_for_team_stats(ownership_data[team])
     dir = os.path.dirname(os.path.realpath(__file__))
     file_to_write = open(f"{dir}/{_OWNERSHIP_FILE}", "w")
-    file_to_write.writelines(json.dumps(processed_data, indent=4))
+    file_to_write.writelines(json.dumps_experimental("apidocs.enable-orjson", processed_data))
```
Comment (anonrig) [inline src/sentry/apidocs/hooks.py:90]: Is there a test that depend on indent=4?

File: src/sentry/backup/crypto.py [hunk 2]
```diff
@@ -135,7 +135,9 @@ def create_encrypted_export_tarball(json_export: json.JSONData, encryptor: Encry
     pem = encryptor.get_public_key_pem()
     data_encryption_key = Fernet.generate_key()
     backup_encryptor = Fernet(data_encryption_key)
-    encrypted_json_export = backup_encryptor.encrypt(json.dumps(json_export).encode("utf-8"))
+    encrypted_json_export = backup_encryptor.encrypt(
+        json.dumps_experimental("backup.enable-orjson", json_export).encode("utf-8")
```
Comment (anonrig) [inline src/sentry/backup/crypto.py:139]: No need for "utf-8" since it is the default value

```suggestion
        json.dumps_experimental("backup.enable-orjson", json_export).encode()
```

File: src/sentry/backup/imports.py [hunk 3]
```diff
@@ -188,7 +188,7 @@ def _import(
             filters.append(user_filter)
 
             # TODO(getsentry#team-ospo/190): It turns out that Django's "streaming" JSON
-            # deserializer does no such thing, and actually loads the entire JSON into memory! If we
+            # deserializer does no such thing, and actually loads_experimental "backup.enable-orjson", the entire JSON into memory! If we
```
Comment (anonrig) [inline src/sentry/backup/imports.py:191]: Is this change intentional? It seems not.

**Review by anonrig — APPROVED** [review]

File: src/sentry/apidocs/hooks.py (hunk 1 above)
Comment (loewenheim) [reply src/sentry/apidocs/hooks.py:90]: AFAICT all these kwargs are actually ignored, but maybe I'm misunderstanding.

File: src/sentry/backup/imports.py (hunk 3 above)
Comment (loewenheim) [reply src/sentry/backup/imports.py:191]: It's not 😅

File: src/sentry/apidocs/hooks.py (hunk 1 above)
Comment (loewenheim) [reply src/sentry/apidocs/hooks.py]: Oh shoot, this is builtin `json`, not `sentry.utils.json`. I'm reverting the `apidocs` part for now.

**Review by anonrig — APPROVED** [review]
